openlibrary/plugins/importapi/import_validator.py

```diff
@@ -1,24 +1,182 @@
+    root: Annotated[str, StringConstraints(pattern=r'^[a-z]{2,3}$')] = Field(
+        ...,
+        description='The MARC21 country code. See https://www.loc.gov/marc/countries/cou_home.html',
+        examples=['enk', 'gw', 'flu'],
+    )
+
+
+class LanguageCode(RootModel):
+    root: Annotated[str, StringConstraints(pattern=r'^[a-z]{3}$')] = Field(
+        ...,
+        description='The MARC21 language code. See https://www.loc.gov/marc/languages/language_code.html',
+        examples=['eng', 'fre', 'ger', 'tha'],
+    )
+
+
+class Isbn10(RootModel):
+    root: Annotated[str, StringConstraints(pattern=r'^([0-9][- ]*){9}[0-9X]$')]
+
+
+class Isbn13(RootModel):
+    root: Annotated[str, StringConstraints(pattern=r'^([0-9][- ]*){13}$')]
+
+
+class WorkKey(RootModel):
+    root: Annotated[str, StringConstraints(pattern=r'^/works/OL[0-9]+W$')]
+
+
+class Language(BaseModel):
+    model_config = ConfigDict(extra="forbid")
+
+    key: Annotated[str, StringConstraints(pattern=r'^/languages/[a-z]{3}$')] = Field(
+        ..., examples=['/languages/eng', '/languages/ger']
+    )
+
+
+class LcClassification(RootModel):
+    root: str = Field(
+        ...,
+        description='The Library of Congress Classification number. See https://www.loc.gov/catdir/cpso/lcc.html We include the imprint date as the last four digits.',  # noqa: E501
+        examples=['BS571.5 .S68 1995', 'Z673.D62 C65 1994'],
+    )
+
+
+class Key(Enum):
+    field_type_link = '/type/link'
+
+
+class Type(BaseModel):
+    model_config = ConfigDict(extra="forbid")
+
+    key: Key
+
+
+class Link(BaseModel):
+    model_config = ConfigDict(extra="forbid")
+    url: AnyHttpUrl
+    title: str
+    type: Optional[Type] = None
+
+
+class EntityType(Enum):
+    person = 'person'
+    org = 'org'
+    event = 'event'
 
 
 class Author(BaseModel):
-    name: NonEmptyStr
+    model_config = ConfigDict(extra="forbid")
+
+    name: NonEmptyStr = Field(examples=['Hubbard, Freeman H.', 'Joan Miró'])
+    personal_name: Optional[str] = Field(
+        None,
+        description="Can be identical to 'name'. TODO: provide information on the intended difference.",
+    )
+    birth_date: Optional[str] = Field(None, examples=[])
+    death_date: Optional[str] = Field(None, examples=[])
+    entity_type: Optional[EntityType] = None
+    title: Optional[str] = Field(None, examples=["duc d'Otrante"])
 
 
 class Book(BaseModel):
+    model_config = ConfigDict(extra="forbid")
+
     title: NonEmptyStr
+    subtitle: Optional[str] = None
     source_records: NonEmptyList[NonEmptyStr]
-    authors: NonEmptyList[Author]
     publishers: NonEmptyList[NonEmptyStr]
-    publish_date: NonEmptyStr
+    authors: list[Author]
+    publish_date: NonEmptyStr = Field(
+        ..., examples=['1992', 'December 1992', '12 January 2002']
+    )
+    location: Optional[NonEmptyList[NonEmptyStr]] = None
+    publish_places: Optional[NonEmptyList[NonEmptyStr]] = None
+    number_of_pages: Optional[float] = None
```

Comment:
`number_of_pages` should probably be an `int` rather than `float`. Is this the auto-conversion of the JSON schema `number`?

Reply:
Yes,indeed